dev/source/docs/using-gazebo-simulator-with-sitl.rst

```diff
@@ -0,0 +1,205 @@
+
+.. tip::
+
+   Gazebo is particularly useful for defining autonomous
+   indoor flights or swarms.
+
+
+Preconditions
+=============
+
+We recommend Ubuntu starting from 14.04.2 (it is working on 16.04 but untest on 16.10) as this is the platform used for testing
+this approach and is also known to be compatible with SITL.
+
+Compiling and installing Gazebo
+===============================
+
+We will be using a standard version of ArduPilot, but a custom fork of Gazebo until gazebo plugin get merge into Gazebo build.
+
+These instructions are derived from http://gazebosim.org/tutorials?tut=install_from_source
+
+Setup your computer to accept software from packages.osrfoundation.org.
+
+::
+
+    sudo sh -c 'echo "deb http://packages.osrfoundation.org/gazebo/ubuntu-stable `lsb_release -cs` main" > /etc/apt/sources.list.d/gazebo-stable.list'
+    wget http://packages.osrfoundation.org/gazebo.key -O - | sudo apt-key add -
+    sudo apt-get update
+    sudo apt-get install build-essential cmake git libboost-all-dev mercurial libcegui-mk2-dev libopenal-dev libswscale-dev libavformat-dev libavcodec-dev  libltdl3-dev libqwt-dev ruby
+    
+Install dependencies
+::
+
+    wget https://bitbucket.org/osrf/release-tools/raw/default/jenkins-scripts/lib/dependencies_archive.sh -O /tmp/dependencies.sh
+    ROS_DISTRO=dummy . /tmp/dependencies.sh
+    sudo apt-get install $(sed 's:\\ ::g' <<< $GAZEBO_BASE_DEPENDENCIES) $(sed 's:\\ ::g' <<< $BASE_DEPENDENCIES)
+
+Remove old Gazebo versions
+::
+
+    sudo apt-get remove '.*gazebo.*' '.*sdformat.*' '.*ignition-math.*'
+    
+Make a gazebo workspace
+::
+
+    mkdir ~/gazebo_ws
+    cd gazebo_ws/
+
+Build and install Ignition Maths
+::
+
+    hg clone https://bitbucket.org/ignitionrobotics/ign-math ~/gazebo_ws/ign-math
+    cd ~/gazebo_ws/ign-math
+    mkdir build
+    cd build
+    cmake .. -DCMAKE_INSTALL_PREFIX=/usr
+    make -j4
+    sudo make install
+
+Build and install Ignition Msgs
+::
+
+    hg clone https://bitbucket.org/ignitionrobotics/ign-msgs ~/gazebo_ws/ign-msgs
+    cd ~/gazebo_ws/ign-msgs
+    mkdir build
+    cd build
+    cmake .. -DCMAKE_INSTALL_PREFIX=/usr
+    make -j4
+    sudo make install
+
+Build and install Ignition Tools
+::
+
+    hg clone https://bitbucket.org/ignitionrobotics/ign-tools ~/gazebo_ws/ign-tools
+    cd ~/gazebo_ws/ign-tools
+    mkdir build
+    cd build
+    cmake .. -DCMAKE_INSTALL_PREFIX=/usr
+    make -j4
+    sudo make install
+
+Build and install SDFormat
+::
+
+    hg clone https://bitbucket.org/osrf/sdformat ~/gazebo_ws/sdformat
+    cd ~/gazebo_ws/sdformat
+    mkdir build
+    cd build
+    cmake .. -DCMAKE_INSTALL_PREFIX=/usr
+    make -j4
+    sudo make install
+
+Build and install Gazebo
+::
+
+    hg clone https://bitbucket.org/osrf/gazebo ~/gazebo_ws/gazebo
+    cd ~/gazebo_ws/gazebo
+    hg checkout ardupilot
+    mkdir build
+    cd build
+    cmake ../
+    make -j4 # NOTE: This will take a long time!
```

Comment:
@khancyr
on the build Gazebo step it fails here:
cmake ../
this is the report
```
-- BUILD WARNINGS
-- HDF5 not found
-- DART not found, for dart physics engine option, please install libdart-core4-dev.
-- Player not found, gazebo plugin for player will not be built.
-- Bullet > 2.82 not found, for bullet physics engine option, please install libbullet2.82-dev.
-- libusb-1.0 not found. USB peripherals support will be disabled.
-- Oculus Rift support will be disabled.
-- GDAL not found, Digital elevation terrains support will be disabled.
-- ronn not found, manpages won't be generated
-- Graphviz not found, Model editor's schematic view will be disabled.
-- OpenSource Virtual Reality (OSVR) support will be disabled.
-- END BUILD WARNINGS
-- Custom CFlags: -std=c++11 -Wall -Wextra -Wno-long-long -Wno-unused-value -Wfloat-equal -Wshadow -Winit-self -Wswitch-default -Wmissing-include-dirs -pedantic -fvisibility=hidden -fvisibility-inlines-hidden -std=c++11
-- Build Type: RelWithDebInfo
-- Install path: /usr/local
-- Configuration successful. Type make to compile gazebo
-- Configuring done
-- Generating done
-- Build files have been written to: /home/parallels/gazebo_ws/gazebo/build
parallels@ubuntu:~/gazebo_ws/gazebo/build$
```